packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -7,26 +12,232 @@ import { IVirtualService } from './virtual-service';
+   */
+  readonly path?: HttpRoutePathMatch;
+
+  /**
+   * Specifies the client request headers to match on. All specified headers
+   * must match for the gateway route to match.
+   *
+   * @default - do not match on headers
+   */
+  readonly headers?: HeaderMatch[];
+
+  /**
+   * The gateway route host name to be matched on.
+   *
+   * @default - do not match on host name
+   */
+  readonly hostname?: GatewayRouteHostnameMatch;
+
+  /**
+   * The method to match on.
+   *
+   * @default - do not match on method
+   */
+  readonly method?: HttpRouteMethod;
+
+  /**
+   * The query parameters to match on.
+   * All specified query parameters must match for the route to match.
+   *
+   * @default - do not match on query parameters
    */
-  readonly prefixPath: string;
+  readonly queryParameters?: QueryParameterMatch[];
+
+  /**
+   * When `true`, rewrites the original request received at the Virtual Gateway to the destination Virtual Service name.
+   * When `false`, retains the original hostname from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+
+  /**
+   * The in-coming request's path to be rewritten when the request is matched.
+   *
+   * @default - the matched part of the path (which is all of it by default) is rewritten to '/'
+   */
+  readonly pathRewrite?: HttpGatewayRoutePathRewrite;
 }
 
 /**
  * The criterion for determining a request match for this GatewayRoute
  */
 export interface GrpcGatewayRouteMatch {
   /**
-   * The fully qualified domain name for the service to match from the request
+   * Create service name based gRPC gateway route match.
+   *
+   * @default - no matching on service name.
    */
-  readonly serviceName: string;
+  readonly serviceName?: string;
+
+  /**
+   * Create host name based gRPC gateway route match.
+   *
+   * @default - no matching on host name.
+   */
+  readonly hostname?: GatewayRouteHostnameMatch;
+
+  /**
+   * Create metadata based gRPC gateway route match.
+   * All specified metadata must match for the route to match.
+   *
+   * @default - no matching on metadata.
+   */
+  readonly metadata?: HeaderMatch[];
+
+  /**
+   * When `true`, rewrites the original request received at the Virtual Gateway to the destination Virtual Service name.
+   * When `false`, retains the original hostname from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+}
+
+/**
+ * Path and prefix properties for HTTP gateway route rewrite.
+ */
+export interface HttpGatewayRoutePathRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route path rewrite.
+   *
+   * @default - rewrite path to '/'.
+   */
+  readonly path?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefixPath?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
```

Comment:
Same here - doesn't matter, but I would call this `prefixPathRewrite`.

Reply:
Addressed